src/PkgDeps.jl

```diff
@@ -82,51 +82,138 @@ reachable_registries(registry_name::String; depots::Union{String, Vector{String}
+            dependencies = find_direct_dependencies(v)
+            if pkg_name in dependencies
+                push!(downstream_dependencies, pkg)
             end
         end
     end
 
     return downstream_dependencies
 end
 
+"""
+    find_direct_dependencies(pkg_name::AbstractString; registries::Array{PkgDeps.RegistryInstance}=reachable_registries())
+
+Find all packages that the latest version of `pkg_name` directly depends on.
+
+# Arguments
+- `pkg_name::AbstractString`: Name of the package
+
+# Keywords
+- `registries::Array{RegistryInstance}=reachable_registries()`: Registries to look into
+
+# Returns
+- `Set{String}`: List of packages `pkg_name` directly depends on
+"""
+function find_direct_dependencies(
+    pkg_name::AbstractString;
+    registries::Array{RegistryInstance}=reachable_registries()
+)
+    upstream_dependencies = Set{String}()
+    for rego in registries
+        if haskey(rego.pkgs, pkg_name)
+            union!(upstream_dependencies, find_direct_dependencies(repo.pkgs[pkg_name]))
+        end
+    end
+    return upstream_dependencies
+end
+
+"""
+    find_direct_dependencies(entry::PkgEntry)
+
+Find all direct dependencies for the latest version of `entry` by parsing the `Deps.toml` file in the registry
+associated to `entry.registry_path`.
+
+# Arguments
+- `entry::PkgEntry`: `PkgEntry` corresponding to the package
+
+# Returns
+- `Array{String}`: List of package names which are the direct dependencies of the latest version of `entry`.
+
+"""
+function find_direct_dependencies(entry::PkgEntry)
+    latest_dependencies = String[]
+    base_path = joinpath(entry.registry_path, entry.path)
+    deps_file_path = joinpath(base_path, "Deps.toml")
+
+    if isfile(deps_file_path)
+        deps_content = parsefile(deps_file_path)
+        dependency_versions = collect(keys(deps_content))
+        latest_version = _get_latest_version(base_path)
+
+        for version_range in dependency_versions
+            if in(latest_version, VersionRange(version_range))
+                dependencies = collect(keys(deps_content[version_range]))
+                append!(latest_dependencies, dependencies)
+            end
+        end
+    end
+    return latest_dependencies
+end
+
+
+"""
+    find_dependencies(pkg_name::AbstractString; registries::Array{RegistryInstance}=reachable_registries())
+
+Find all packages that the latest version of `pkg_name` depends on (directly or indirectly).
+
+# Arguments
+- `pkg_name::AbstractString`: Name of the package
+
+# Keywords
+- `registries::Array{RegistryInstance}=reachable_registries()`: Registries to look into
+
+# Returns
+- `Set{String}`: List of packages which `pkg_name` depends on.
+"""
+function find_dependencies(
+    pkg_name::AbstractString;
+    registries::Array{RegistryInstance}=reachable_registries()
+)
+    return find_dependencies!(Set{String}(), pkg_name; registries)
+end
+
+# recursive helper that accumulates into `upstream_dependencies`
+function find_dependencies!(
```

Comment:
Maybe prefacing this with an underscore to denote that it's an "internal function"?